### PR TITLE
feat: add ent plugin

### PR DESCRIPTION
**Pull request by xiaohuo473225193**
feat: add ent plugin, https://github.com/ent/ent

### CODE REVIEW

**Comment by wu-sheng** [comment]
Thanks, the tests are passed now. Let's wait for @arugal review.

File: ent/ent.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	apmDB.DB.SetMaxIdleConns(50)
+	apmDB.DB.SetMaxOpenConns(200)
+
+	drv.Conn = entsql.Conn{ExecQuerier: apmDB}
+
+	EntClient := entschema.NewClient(
+		entschema.Driver(
+			NewDriver(drv, apmDB),
+		),
+	)
```
Comment (arugal) [inline ent/ent.go:89]: @xiaohuo473225193 The `gen/entschema` seems to be generated based on [schema](https://github.com/SkyAPM/go2sky-plugins/pull/52/files#diff-c5fc389a809e819c5217697a1711b202bfb7f5594a14da80cdabad8b8c6a939cR28-R30),  this plugin does not apply to other schema, right？

Comment (xiaohuo473225193) [reply ent/ent.go:89]: No. The plugin is used for all schema definitions.

Comment (arugal) [reply ent/ent.go:89]: But, The [NewClient](https://github.com/SkyAPM/go2sky-plugins/pull/52/files#diff-22e40780487774f76e02ccbc532177f7480a6b484b1dcfddb1fe2d5ed4145aa3R42-R53) that you have here can only be used for the [user schema](https://github.com/SkyAPM/go2sky-plugins/pull/52/files#diff-c5fc389a809e819c5217697a1711b202bfb7f5594a14da80cdabad8b8c6a939c) that you define.

Comment (xiaohuo473225193) [reply ent/ent.go:89]: Ent can only operate based on the code it generates. NewClient it's the only way in method.

Comment (xiaohuo473225193) [reply ent/ent.go:89]: `NewClient` has nothing to do with the schema definition.

Comment (arugal) [reply ent/ent.go:89]: > Ent can only operate based on the code it generates. NewClient it's the only way in method.

When I create a new schema, I need to regenerate the code, right?

Comment (xiaohuo473225193) [reply ent/ent.go:89]: Yes. But schema change affects only the schema model. The Ent code itself can be considered fixed.

Comment (arugal) [reply ent/ent.go:89]: Can you demonstrate how to reference ent plugin in the new repository?

Comment (xiaohuo473225193) [reply ent/ent.go:89]: This is actually an example, not a specific plugin. This is just one way to access the trace method.

Comment (xiaohuo473225193) [reply ent/ent.go:89]: It is just a reference method provided by Ent access trace. I use this approach in my project.

Comment (hoslo) [reply ent/ent.go:89]: why stop？

**Pull request closed without merging**